### ISSUE TITLE
Environment variable not being resolved when sending request

### ISSUE THREAD

**Issue by idevtech** [a85/POSTMan-Chrome-Extension#6001]
v7.02 Windows

Description
I have an environment variable "urlApi" defined w/ a value of "http//localhost:5050". The environment variable is then used in a request as {{urlApi}}. When the request is sent the follow error message is displayed indicating the variable is not being resolved to the proper url. All of the requests worked w/ this env variable until Postman was upgraded to the current version.

_There was an error connecting to %7B%7BurlApi%7D%7D/api/v1/geoserver/inventorypoints?extent.x1=-7902186.581849624&extent.y1=5178452.208557574&extent.x2=-7899324.970603597&extent.y2=5180121.8798939455._

**Comment by harryi3t** [comment]
@idevtech Can you a screenshot showing what you see when you hover on the variable?
I have attached a sample screenshot.
![image](https://user-images.githubusercontent.com/5207331/53617599-d2c6dc00-3c0d-11e9-8c28-2d1bb92f150e.png)

Also, try switching to a different environment to see if the problem is with a particular environment or all of them.

**Comment by idevtech** [comment]
I did some experimentation this morning and after deleting and re-creating
the env variable it works correctly. The variable had been in place for a
long time, across multiple version upgrades, so I wonder if the storage
format or something related changed which caused the problem.

Chris

**Comment by a85** [comment]
Closing as this seems to be working now.

**Comment by NadavOs90** [comment]
this happens to me in 7.0.9

**Comment by MarlieChiller** [comment]
experiencing this on macOS with Version 7.5.0 (7.5.0)

**Comment by sambyers** [comment]
Similar issue here. I created an environment with variables, imported a collection, and couldn't use the environment variables with requests in the collection. I deleted the environment and recreated it and now it works.

macOS 10.14.6
Postman 7.5.0

**Comment by ziedjebali** [comment]
> Similar issue here. I created an environment with variables, imported a collection, and couldn't use the environment variables with requests in the collection. I deleted the environment and recreated it and now it works.
> 
> macOS 10.14.6
> Postman 7.5.0

Same thing happened to me and this worked as a fix

macOS 10.15.2
Postman v7.16.0

**Comment by zhitomir-oreshenski-mm** [comment]
Hello everyone,

Thank you guys, I've been struggling with that yesterday. 

You have to basically delete the current environment(develop, prod) and try to recreate it with new env variables.

**Comment by ScottCarr73** [comment]
This issue is getting annoying.  When I start Postman, I basically have to delete and readd each of the variables my requests are using, before the request will work.  Today, the QA person on the team, had to delete and readd the same variables, so he could get the test to work for the QA process.

This should be reopened, as the "Fix" is to delete and readd the variable.  Why are the variables not being recognized?  They ARE in the list, changing the variable and re-saving, doesn't fix the issue.  Only delete and readd of the variable, lets the requests see the variable again.  It appears that this is ONLY for the person specifically running in that session as well.  Others that are using the same request, have to go through the process, each time they restart Postman.

**Comment by ahyldig** [comment]
Can confirm, I too am struggling with this issue EVERY DAY. Quite mind-boggling that this has not been fixed as of yet.

To elaborate a bit further, this happens when I open Postman (after a complete computer shutdown). I then have to manually change the value in an existing environment variable for it to work again (use the actual value). This seems like some sort of caching/memory issue where the environment variables are not loaded upon startup of the Postman desktop App.

**Comment by vincelarkin** [comment]
Not sure if this works for everybody, but in my case I had to go to enviroments and click on the v icon beside my custom set of variables. 

![image](https://user-images.githubusercontent.com/16820349/113686382-7cb7a300-96c7-11eb-9d49-321905a29ceb.png)

It was somehow turned off, and clicking it fixed the issue for me.

**Comment by mikebrant-xdelivery** [comment]
+1 on having this same issue. I had double-checked my active environment as well.  I had to delate and re-add variables for them to be picked up :(

**Comment by timbochamp** [comment]
The latest comments since this issue being closed look similar to #9729 if you'd like to follow the resolution there! Apologies for the frustration in having to find workarounds for resolving these variables in the meantime.

**Comment by Trapsta** [comment]
duplicating the environment and switching to the new copy works in my case (Mac.  10.14.6 , Postman 8.5.1). I really wanted to avoid keying in the values all over again 😉

**Comment by cw-dos** [comment]
Same issue, struggled with this for hours, I thought I was going crazy. The fix for me was deleting the environment entirely and re-adding the variables and magically they worked.

**Comment by Cyral** [comment]
+1, I used environments for the first time today and couldn't figure out why variables weren't being picked up. Re-created it and it works.

**Comment by EagleVee** [comment]
Still happening on MacOS, version 9.9.3

**Comment by shubhbhargav** [comment]
@EagleVee In order to debug further, can you please share the steps to reproduce it? (preferably a screenshot / video where you have selected the environment with the variable but it's failing when you send the request)

**Comment by deepumukundan** [comment]
I had the same issue on macOS client 9.10.1 of Postman. After some troubleshooting, deleting and creating variables etc I think it is because of how the environment variable is saved the first time is causing this issue.

See screenshot below
1. Test1: Create env var, Assign an initial value, save - No issues, also Initial value is copied to current value
2. Test2: Redid step 1
3. Test3: Create env var, save, then assign initial value and then save - Causes the issue mentioned above, no current value set
![Screen Shot 2022-01-25 at 2 26 17 PM](https://user-images.githubusercontent.com/1459484/151070601-c4d93631-df3d-40e7-9aee-9a7459d5eebc.png)

Since I did not care about initial vs current value and was mostly focusing on using the variable later, I opted for use case 1 among the above to get me unblocked. But I believe the issue is that current value not being set (as in use case 3), is causing Postman to not pick the initial value as well for some reason which was set later by the user.

**Comment by fidesachates** [comment]
this needs to be reopened. Having this issue now.

**Comment by BonnieThorne** [comment]
I am having a slightly different issue. If I put a persistent value in my environment, in this case my url, it will be picked up in my GET request, however for my tokens, I have a prerequest script that swaps the authentication token every iteration and assigns it to a key called token with a value left blank. This has worked in plenty of my other collections. However, I did originally receive these files from a colleague with an outdated version, so I remade the collection and environment to be sure it was not that problem, but the value is still not being grabbed from the environment. I know my prerequest script is working because I have it logging the value of "token" in the environment every iteration (see screenshot. real tokens are blacked out)
![image](https://user-images.githubusercontent.com/74556318/152436720-8f2970c9-4002-49dc-8cf4-5fb8f41f1262.png)
![image](https://user-images.githubusercontent.com/74556318/152437396-4d851cf7-ebd8-4b86-ad46-2c30a107aff1.png)
![image](https://user-images.githubusercontent.com/74556318/152437511-5955f3af-d5ff-4271-8d35-0eb93062f408.png)

**Comment by malvikach** [comment]
@deepumukundan @fidesachates This is an expected behavior wrt initial and current values. Thanks for your feedback on this though. We would love to talk to you and know more about your use cases though and understand what problems you are facing while working with environments. Let me know if you would be interested. Feel free to drop an email to me at malvika.chaudhary@postman.com or provide your details [here](https://forms.gle/w5RWEEM2wNCwCXt58). I will coordinate a time to speak with you.

**Comment by malvikach** [comment]
Hi All,

We are going to track the issue (wrt variable not getting resolved while sending a request) on this ticket: https://github.com/postmanlabs/postman-app-support/issues/10644

Please feel free to provide additional information to reproduce this issue on the new ticket.

**Comment by ghost** [comment]
I just met this problem as well, can try exporting the variables into a json file, then inspect the json file to find if you add some whitespace around the variable names.

**Comment by ZissisT** [comment]
Had same issue, but https://github.com/postmanlabs/postman-app-support/issues/6001#issuecomment-813957545 was the fix. You have to set your environment as active environment (check if you have checkbox next to it in the environment list)

**Comment by pamelawilson65** [comment]
This issue it *not* resolved. I've tried every suggestion in this thread and nothing has fixed the issue. 
1) Made sure the v is selected under Environments.
2) Deleted all variables and offending environment and  restarted PM and reimported ENV (variable is created, but not populated)
3) removed collections, closed all tabs, delete offending variables, restarted PM

PM tells me I have the latest version 9.17.1. Most of my team is not having this problem, b/c they do not use the connected online version of PM. So I suspect it's a problem with storing the data.  If anyone has any suggestions, please let me know. I'm completely blocked and spent half my day on this PM (clearly untested product) problem. What alternatives do we have to PM?  This has happened 2-3 dozen times for me, in the last couple years, but was always able to fix it with restarting PM or removing offending variables, this is the first time it's nots fixable, after 3-4h of investigations. How do I explain this lost time to my boss??? :-(

**Comment by shubhbhargav** [comment]
@pamelawilson65 We understand the frustration. Can you please share your system details and a screen recording of the issue? We will get it resolved ASAP for you.

**Comment by LarisaMocu** [comment]
I'm having the same problem. Trying to save environments variables, with pm.environment.set(), and it's not working. 
Example: I'm setting "token" variable, pm.environment.set("token", pm.response.text()), and I see the "token" variable in the environment, but it's empty. If I run the entire collection (right-click -> Run collection) it works fine, but I really need to run requests individually. Please help

**Comment by malvikach** [comment]
@LarisaMocu Could you try console.log(pm.response.text()) to check the value in this field before setting it in the `token` variable in the single request. When the request is executed, you can then [check postman console](https://learning.postman.com/docs/sending-requests/troubleshooting-api-requests/) to see the logs. 

It is possible that the response is empty due to which you see an empty value assigned to the environment variable. 
Let me know if this helps.

**Comment by sundersham** [comment]
I was also struggling with the same issue. Windows 10, Postman v9.21.2. 
Now, I'm able to resolve it. Details given below:

The root cause in my case was the extra forward slash given at the end of the url. Removing the slash resolved the issue.
It was working fine with the previous versions of Postman. The issue start after auto upgrading to the new version.

fin_api_url = https://xxx/azure/live/finance//
fin_api_url = https://xxx/azure/live/finance/

**Comment by suntoshakula** [comment]
the issue still exists,currently my enviroment variables not being read by monitor requests.

**Comment by LarisaMocu** [comment]
I don’t know if this makes any sense, but what worked for me was to sign out/sign back in from my Postman account.

**Comment by suntoshakula** [comment]
thanks so much, i cleared the app properly with unwanted collections, sign out and sign in boom it works now. Awesome larisamocu you the saviour man

**Comment by suntoshakula** [comment]
@larisamocu, what you suggested was right. i did signout and sign in and what I did additional was to click on persist all button to resolve the issue. now I am able to share across different machines + it shows success message.

Thanks again for your help.

**Comment by mcnallyjames** [comment]
Hello I am now having this issue: 
Version 9.21.4
Steps to reproduce:
Using Enviroment set the inital value and/or current value (For the below screenshots I have done both however tried all combinations of the variable value)
Click Save (IMG 1)
Open a Post Request within the collection on the workspace and check value of Enviroment Variable (IMG 2)
Send the request, check console once error occurs (IMG 3 & 4)
Check the value of variable and it is returned as null (IMG 5)

Please note I have had to cover up sensitive information.

Steps taken to resolve:
Log out and back in to Postman (Not linked to google account)
Deleted the Enviroment and recreated it with the same variables.
Replace the variable with a hardcoded value, this does return a correct result.

Note: Using a different enviroment but with the same variable, though a different value being stored, does return a successful test.

![IMG 1](https://user-images.githubusercontent.com/32570037/175267123-84d267df-0c56-4c29-b72d-71e774786150.jpg)
![IMG 2](https://user-images.githubusercontent.com/32570037/175267129-c248499d-d36d-4ee1-a5a2-93ba5eb82439.jpg)
![IMG 3](https://user-images.githubusercontent.com/32570037/175267130-1cd0475a-618d-4917-a02f-83d362bea7c0.jpg)
![IMG 4](https://user-images.githubusercontent.com/32570037/175267114-3775cf83-a33b-4dfd-8a52-61b7681017bf.jpg)
![IMG 5](https://user-images.githubusercontent.com/32570037/175267722-b747d5ec-10fe-4f02-a46e-7f63067688bb.jpg)

**Comment by suntoshakula** [comment]
do not type in enviroment variable: "initial value" let the persist option fill it for you(that's what I did). but before logout all instances and login. sometimes restarting machine is beneficial too.
also make sure when you login to another instance the values are still existing there also sometimes it is empty not sure why they don't sink it could be because they are not persisted (my guess)

make sure to signout and sign in check the environment variable make sure all values are passed using the console option statement, I think do not post so many details it will be confusing just target the problem. please let me know how it goes.

this is a very weird @postman-support (postman setup) I am using free version may be that is the reason it does that.

**Comment by mcnallyjames** [comment]
> do not type in enviroment variable: "initial value" let the persist option fill it for you(that's what I did). but before logout all instances and login. sometimes restarting machine is beneficial too. also make sure when you login to another instance the values are still existing there also sometimes it is empty not sure why they don't sink it could be because they are not persisted (my guess)

> this is a very weird @postman-support (postman setup) I am using free version may be that is the reason it does that.

Hi,
I have attempted all these and the issue remains across all my enviroments where I have set the intial value, the current value is not being populated as part of the test run. I have set the values to persist all and it returns:
POST http://nullv1/Account/
Error: getaddrinfo ENOTFOUND nullv1

The only workaround I have is to set the values in a pre request script before each endpoint and that seems to work.

**Comment by mcnallyjames** [comment]
I seem to have found a workaround which has worked.
Take the value stored in inital value, copy it and put it in Current Value > Click Persist All > Click Save > Run the request and the value from Current is stored correctly and results are returned.

**Comment by suntoshakula** [comment]
yep thats what i meany by click on persist because it copies from current to initial value when you click persist. glad it worked for you because it worked for me before too. take care

**Comment by AnotherWayIn** [comment]
still the same issue on latest mac version

**Comment by preethammavin** [comment]
Initial values are synced to the servers, and current values are in the local machine session. With persist all - click we copy the value which is in the machine and persist in the environment and is accessible via the cloud. You can also click on reset all - which copies all the initial values into the current value. We do not auto-reset the current values in the local machines, as the users can build workflows using the values incrementally

**Comment by bcstryker** [comment]
Instead of everyone posting their own message, cluttering the issue, please just give this message a thumbs up if you're still experiencing this issue (I am).

**Comment by dstromberg** [comment]
I'm having this problem too, on:
Postman v10.10.6

$ sw_vers
cmd output started 2023 Fri Mar 03 03:42:41 PM PST
ProductName:		macOS
ProductVersion:		13.1
BuildVersion:		22C65

**Comment by pranavmappoli** [comment]
Still facing this issue, very poor to see this opened for a long and not able to fix

**Comment by ngnguyen1** [comment]
5 years and still facing with this. It get initial value instead current value in environment. :(

**Comment by ayamgoreng47** [comment]
so, whats the mitigation? any help please? i've similar problem like this too

**Comment by mcnallyjames** [comment]
The workaround is to clear the initial value, put the value you want in
current value then click persist all then save, that'll fix the problem,
which is intermittent so difficult to reproduce

**Comment by darman12** [comment]
Want to chime in that I'm having this issue, as well. @mcnallyjames' workaround did not work for me, unfortunately.

EDIT: The environment variables that weren't resolving when sending the request were set to Type "secret". Changing them to "default" got me around the issue but that doesn't seem like a good solution.

**Comment by AbrahamBrookes** [comment]
I had this issue this morning on a newly created environment. Turned out I had to click 'save' after editing the variables:
![image](https://github.com/postmanlabs/postman-app-support/assets/8340776/690ba26c-5229-4ca1-af5a-661cb7f8f25b)

**Comment by xrm-pro** [comment]
To resolve this I duplicated the entire variable set and then used the copy.

**Comment by ben-eb** [comment]
Running into this issue today with an environment one of my team members created. One of the environment variables has an initial value (set by my colleague, due to it being non-sensitive), and the other variables have current values (which I set). These variables cannot be resolved by the collection authentication.

It seems that if any environment variable is set as initial Postman prefers the initial value for _all_ variables, rather than checking the current variable _for each variable_ and falling back to initial (arguably what it should default to).

This is using Postman 11.7.0

**Comment by Vidrg** [comment]
> I had this issue this morning on a newly created environment. Turned out I had to click 'save' after editing the variables: ![image](https://github.com/postmanlabs/postman-app-support/assets/8340776/690ba26c-5229-4ca1-af5a-661cb7f8f25b)

worked

**Comment by MarkDonovan51** [comment]
Issue occurred for me today. Workaround was sufficient. It would be nice to re-open this issue and fix it.

**Comment by DSchau** [comment]
@MarkDonovan51 have you updated to the latest version of Postman? We've actually made it so that environments are autosaved by default.

![autosave by default](https://blog.postman.com/engineering/wp-content/uploads/2025/09/variables-toggle-column.gif)

https://blog.postman.com/engineering/a-simpler-and-local-first-variables-experience/

**Comment by MarkDonovan51** [comment]
Yes! I've experienced the improvements that are being made and appreciate them, thank you. It still occurred for me. I believe I'm on the latest version, v11.68.5, though I believe I was on the previous release when I fixed the issue yesterday by duplicating the environment variables, deleting the original, and re-naming the duplicate.

<img width="279" height="281" alt="Image" src="https://github.com/user-attachments/assets/b03aa852-1e3b-405b-9e6f-f17d31f62da5" />
